### ISSUE TITLE
Distribute an AppImage

### ISSUE THREAD

**Issue by xtaniguchimasaya** [tectonic-typesetting/tectonic#280]
This issue is related with #185 #140.
Now I'm making an [AppImage](https://appimage.org) of tectonic. And I can show you my recipe because I have an appimage of tectonic. However, AppImage requires a package to have `tectonic.desktop` and `tectonic.png`. Currently I use a dummy icon to make a sample package but I'd like to use **officially** icon for the package.  Do you have any plan to make icon. Thank you.

```shell
$ cargo build --release
$ ./linuxdeploy-x86_64.AppImage --icon-file=./tectonic.png --desktop-file=./tectonic.desktop --appdir=./target/release --executable=./target/release/tectonic --output=appimage
```

**Comment by pkgw** [comment]
@xtaniguchimasaya Thank you for making that! Feel free to use it in your AppImage package for now. You have inspired me to see if I can hire a professional designer to come up with an "official" logo (I am sure that there are lots of non-obvious best practices involved) but I don't want your work to be stuck on anything like that.

**Comment by xtaniguchimasaya** [comment]
Okay, I use that icon until making up an official logo. Next, I have already had an AppImage of this project, as I said. AppImage has no central repository. It means that these AppImage is preferred to distribute from upstream themselves. Of course, I can make the repository to use GitHub Releases for users, but I'd like to upload them to `tectonic-typesetting/tectonic-appimage` repository if you accept. For example, Krita that is drawing software from KDE distributes the AppImage in their project website. It expresses good to all users because that is *latest* and *nearest* from upstream/HEAD. 
Thirdparty repositories are also used frequently but the lifetime of the repository is unknown  regardless of the lifetime of upstream project. How about this proposal?

**Comment by probonopd** [comment]
> Now I'm making an AppImage of tectonic. 

Can the AppImage be downloaded somewhere for testing? Thanks.

**Comment by xtaniguchimasaya** [comment]
@probonopd Now I upload recipe and binary to GitHub. 

https://github.com/xtaniguchimasaya/tectonic-appimage

@pkgw I'm happy to hear that. I guess we can put the build script for AppImage in `dist/`, but it uses git submodule to build AppImage. I try to use parent directory to build AppImage. Then I send it as pull request.

**Comment by pkgw** [comment]
OK, this is merged! I made an AppImage for the 0.1.11 release and uploaded it as a GitHub release artifact. The link is <https://github.com/tectonic-typesetting/tectonic/releases/download/v0.1.11/Tectonic-0.1.11-x86_64.AppImage>

**Comment by probonopd** [comment]
Great @pkgw. There are still some minor issues with the AppImage that currently make it fail the automated test, should be possible to be fixed easily. Please see https://github.com/AppImage/appimage.github.io/pull/1025#issuecomment-450580552.
